src/bootstrap/setup.rs

```diff
@@ -107,6 +108,19 @@ pub fn setup(src_path: &Path, profile: Profile) {
     let include_path = profile.include_path(src_path);
     println!("`x.py` will now use the configuration at {}", include_path.display());
 
+    let build = TargetSelection::from_user(&env!("BUILD_TRIPLE"));
+    let stage_path = ["build", build.rustc_target_arg(), "stage1"].join("/");
+    let toolchain_link =
+        Command::new("rustup").args(&["toolchain", "link", "stage1", &stage_path[..]]).output();
+    if toolchain_link.is_err() {
```

Comment:
This is checking the wrong thing - `output()` only fails if the syscall itself fails, not if the status is non-zero.
```suggestion
    if toolchain_link.is_err() || !toolchain_link.status().success() {
```